src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -244,18 +253,31 @@ public void closeAll() {
     closeStreamingClient();
   }
 
+  /**
+   * This function is called during rebalance. Please ensure we dont wipe off state when we are
+   * invoking this function. i.e after rebalance, we would still want to preserve metadata like
+   * streamingBuffer, processedOffset
+   *
+   * <p>All the channels are closed. The client is still active. Upon rebalance, (inside {@link
+   * com.snowflake.kafka.connector.SnowflakeSinkTask#open(Collection)} we will reopen the channel.
+   *
+   * @param partitions a list of topic partition
+   */
   @Override
   public void close(Collection<TopicPartition> partitions) {
     partitions.forEach(
         topicPartition -> {
-          String partitionChannelKey =
+          final String partitionChannelKey =
               partitionChannelKey(topicPartition.topic(), topicPartition.partition());
-          LOGGER.info("Closing partition channel:{}", partitionChannelKey);
-          partitionsToChannel.get(partitionChannelKey).closeChannel();
-          partitionsToChannel.remove(partitionChannelKey);
+          TopicPartitionChannel topicPartitionChannel =
+              partitionsToChannel.get(partitionChannelKey);
+          topicPartitionChannel.closeChannel();
```

Comment:
We would need to handle the potential exceptions thrown during close channel, we could just catch it and ignore it because we will start with the latest committed offset

Reply:
Sounds good, let me do that..thanks

Reply:
Actually, I already catch that in `closeChannel()`